services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/DiagnosisKeyBundlerTest.java

```diff
@@ -0,0 +1,168 @@
+package app.coronawarn.server.services.distribution.assembly.diagnosiskeys;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import app.coronawarn.server.common.persistence.domain.DiagnosisKey;
+import app.coronawarn.server.services.distribution.config.DistributionServiceConfig;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.context.ConfigFileApplicationContextInitializer;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@EnableConfigurationProperties(value = DistributionServiceConfig.class)
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {DistributionServiceConfig.class},
+    initializers = ConfigFileApplicationContextInitializer.class)
+class DiagnosisKeyBundlerTest {
+
+  @Autowired
+  DistributionServiceConfig distributionServiceConfig;
+
+  @Test
+  void testEmptyListWhenNoDiagnosisKeys() {
+    DiagnosisKeyBundler bundler = new DiagnosisKeyBundler(List.of(), distributionServiceConfig);
+    assertThat(bundler.getDiagnosisKeysDistributableAt(LocalDateTime.of(1970, 1, 1, 0, 0, 0))).hasSize(0);
+  }
+
+  @Test
+  void testEmptyListWhenGettingDistributableKeysBeforeEarliestDiagnosisKey() {
+    List<DiagnosisKey> diagnosisKeys = buildDiagnosisKeys(6, 50L, 5);
+    DiagnosisKeyBundler bundler = new DiagnosisKeyBundler(diagnosisKeys, distributionServiceConfig);
+    assertThat(bundler.getDiagnosisKeysDistributableAt(LocalDateTime.of(1970, 1, 1, 0, 0, 0))).hasSize(0);
+  }
+
+  @Nested
```

Comment:
This might not be the best use case for ``@Nested`` as there is only one share resource and no shared setUp/tearDown code. The only benefit (?) seems to be that we have one large file instead of two smaller ones.